src/core/ext/transport/chttp2/transport/writing.cc

```diff
@@ -75,6 +75,7 @@ static void maybe_initiate_ping(grpc_chttp2_transport* t) {
     }
     return;
   }
+  grpc_core::ExecCtx::Get()->InvalidateNow();
```

Comment:
I'm not sure about this one -- @yashykt should review this one.

Reply:
This does run within a combiner if that's what the original question is

Reply:
The more I think about this, the more I'm thinking that we should either not cache `Now()` at all or maybe we should add `InvalidateNow()` at some more general locations, for example, when closures are executed on the work serializer/combiner